packages/dataprovider/src/buildDataProvider.ts

```diff
@@ -2,12 +2,17 @@ import merge from "lodash/merge";
 import buildRaGraphqlDataProvider from "ra-data-graphql";
 import { DELETE, DELETE_MANY, UPDATE, UPDATE_MANY } from "react-admin";
 import { buildQueryFactory } from "./buildQuery";
-import { Options } from "./types";
+import { Options, OurOptions } from "./types";
 
 import { makeIntrospectionOptions } from "./utils/makeIntrospectionOptions";
 
+export const defaultOurOptions: OurOptions = {
+  queryDialect: "nexus-prisma", // todo woj would be nice if it would be the type, maybe enum?
```

Comment:
haha we actually talked about this ;-)

its a union type. Unless you want to iterate over values, an union type is (imho) often easier than an enum, because you don't have to know the import, but type-checking and auto-complete still work, so slightly better ergonomics.

Union types are quite common in functional languages. In "real" functional languages, they are even more powerful and amazing to use as you can pattern match against them even if they are structurally very different. In typescript, you can also pattern-match, but only within whats possible on runtime in js (e.g. strings are no problem).

Reply:
You are right, I just was missing the autocomplete.